src/pages/workspace/content/up-review/index.tsx

```diff
@@ -103,8 +105,24 @@ const UpForReviewPage = () => {
       try {
         setLoading(true);
         const query = debouncedSearchTerm || "";
-        const offset =debouncedSearchTerm!==""? 0 : page * LIMIT;
-        const primaryCategory = filter.length ? filter : [];
+        let offset =debouncedSearchTerm!==""? 0 : page * LIMIT;
+        const localSelectedFilters= localStorage.getItem("selectedFilters");
+        const selectedFilters = localSelectedFilters?JSON.parse(localSelectedFilters ): null;
+        const primaryCategory = selectedFilters? selectedFilters : filter.length ? filter : [];
+ 
+        if (prevFilterRef.current !== filter) {
+          offset=0;
+          setPage(0);
+          router.push(
+            {
+              pathname: router.pathname,
+              query: { ...router.query, page: 1 }, 
```

Comment:
@AkshataKatwal16 everywhere same code repeated, can u optimize this... and format files

Reply:
@itsvick should i write common function for that?
```
if (prevFilterRef.current !== filter) {
          offset=0;
          setPage(0);
          router.push(
            {
              pathname: router.pathname,
              query: { ...router.query, page: 1 }, 
            },
            undefined,
            { shallow: true } 
          );
          prevFilterRef.current = filter;
        }
```

Reply:
@itsvick moved that router logic to common component  searchbox